src/diffusers/models/attention.py

```diff
@@ -187,95 +187,137 @@ def __init__(
         dim: int,
         num_attention_heads: int,
         attention_head_dim: int,
-        dropout=0.0,
+        dropout: Optional[float] = 0.0, 
         cross_attention_dim: Optional[int] = None,
         activation_fn: str = "geglu",
         num_embeds_ada_norm: Optional[int] = None,
         attention_bias: bool = False,
-        only_cross_attention: bool = False,
         upcast_attention: bool = False,
+        temb_channels: Optional[int] = None, # for ada_group_norm
+        attn1_type: str = 'cross',  # cross, self
+        attn2_type: Optional[str] = None, # None, cross
+        with_ff: bool = True,
+        use_conv_proj: bool = False,
+        cross_attention_norm: bool = False,
+        attention_dropout: Optional[float] = None, 
+        group_size: int = 32   
     ):
         super().__init__()
-        self.only_cross_attention = only_cross_attention
+
+        self.attn1_type = attn1_type
+        self.attn2_type = attn2_type
         self.use_ada_layer_norm = num_embeds_ada_norm is not None
+        self.use_ada_group_norm = temb_channels is not None
+        
+        if self.use_ada_layer_norm:
+            self.norm1 = AdaLayerNorm(dim, num_embeds_ada_norm)
+        elif self.use_ada_group_norm:
+            self.norm1 = AdaGroupNorm(temb_channels, dim, max(1, dim // group_size))
+        else:
+            self.norm1 = nn.LayerNorm(dim)
 
-        # 1. Self-Attn
         self.attn1 = CrossAttention(
             query_dim=dim,
             heads=num_attention_heads,
             dim_head=attention_head_dim,
             dropout=dropout,
             bias=attention_bias,
-            cross_attention_dim=cross_attention_dim if only_cross_attention else None,
+            cross_attention_dim=cross_attention_dim if attn1_type == 'cross' else None,
             upcast_attention=upcast_attention,
+            use_conv_proj= use_conv_proj,
+            cross_attention_norm = cross_attention_norm if attn1_type == 'cross' else None,
+            attention_dropout = attention_dropout, 
         )
-
-        self.ff = FeedForward(dim, dropout=dropout, activation_fn=activation_fn)
+        
+        if with_ff:
```

Comment:
Interesting - the upscaler does not use a feed-forward layer? Usually attention always uses:
self-attention
(maybe cross-attention)
feed-forward

If it's not the case here we should maybe think actually about a whole new class